logdna/data_source_alert_test.go

```diff
@@ -0,0 +1,218 @@
+					resource.TestCheckResourceAttr(data, "pagerduty_channel.0.%", "6"),
+					resource.TestCheckResourceAttr(data, "pagerduty_channel.0.immediate", "false"),
+					resource.TestCheckResourceAttr(data, "pagerduty_channel.0.operator", "presence"),
+					resource.TestCheckResourceAttr(data, "pagerduty_channel.0.terminal", "true"),
+					resource.TestCheckResourceAttr(data, "pagerduty_channel.0.triggerinterval", "15m"),
+					resource.TestCheckResourceAttr(data, "pagerduty_channel.0.triggerlimit", "15"),
+				),
+			},
+		},
+	})
+}
+
+func TestDataSourceAlert_basicWebhook(t *testing.T) {
+	data := "data.logdna_alert.remote"
+	rsName := "test"
+
+	args := []string{rsName, webhook}
+	resource.Test(t, resource.TestCase{
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testDataSourceAlertConfig(args...),
+				Check: resource.ComposeTestCheckFunc(
+					testDataSourceAlertExists(data),
+					resource.TestCheckResourceAttr(data, "name", rsName),
+					resource.TestCheckResourceAttr(data, "webhook_channel.#", "1"),
+					resource.TestCheckResourceAttr(data, "webhook_channel.0.%", "9"),
+					resource.TestCheckResourceAttr(data, "webhook_channel.0.headers.%", "2"),
+					resource.TestCheckResourceAttr(data, "webhook_channel.0.headers.hello", "test3"),
+					resource.TestCheckResourceAttr(data, "webhook_channel.0.headers.test", "test2"),
+					resource.TestCheckResourceAttr(data, "webhook_channel.0.bodytemplate", "{\n  \"fields\": {\n    \"description\": \"{{ matches }} matches found for {{ name }}\",\n    \"issuetype\": {\n      \"name\": \"Bug\"\n    },\n    \"project\": {\n      \"key\": \"test\"\n    },\n    \"summary\": \"Alert From {{ name }}\"\n  }\n}"),
+					resource.TestCheckResourceAttr(data, "webhook_channel.0.immediate", "false"),
+					resource.TestCheckResourceAttr(data, "webhook_channel.0.method", "post"),
+					resource.TestCheckResourceAttr(data, "webhook_channel.0.url", "https://yourwebhook/endpoint"),
+					resource.TestCheckResourceAttr(data, "webhook_channel.0.terminal", "true"),
+					resource.TestCheckResourceAttr(data, "webhook_channel.0.triggerinterval", "15m"),
+					resource.TestCheckResourceAttr(data, "webhook_channel.0.triggerlimit", "15"),
+				),
+			},
+		},
+	})
+}
+
+func TestDataSourceAlert_multipleChannels(t *testing.T) {
+	data := "data.logdna_alert.remote"
+	rsName := "test"
+
+	args := []string{rsName, "multiple"}
+	resource.Test(t, resource.TestCase{
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testDataSourceAlertConfig(args...),
+				Check: resource.ComposeTestCheckFunc(
+					testDataSourceAlertExists(data),
+					resource.TestCheckResourceAttr(data, "name", rsName),
+				),
+			},
+		},
+	})
+}
+
+func testDataSourceAlertConfig(args ...string) string {
+	name := args[0]
+	integration := args[1]
+	isMultiple := false
+	if integration == "multiple" {
+		isMultiple = true
+	}
+
+	var sb strings.Builder
+	fmt.Fprintf(&sb, pcStr+"\n", serviceKey)
```

Comment:
NIT: I think the line feed char is unnecessary as consts already end up with a LF char :)